client/src/views/Simulation/Simulation.vue

```diff
@@ -10,9 +10,11 @@
         Search
       </button>
 
-      <button class="btn btn-primary">
+      <button
+        class="btn btn-primary"
+        @click="onToggleProvenanceGraph">
         <font-awesome-icon :icon="['fas', 'project-diagram' ]" />
-        Provenance Graph
+        <span>{{ isProvenanceGraphOpen ? 'Close Provenance Graph' : 'Open Provenance Graph' }}</span>
```

Comment:
You can avoid repetition here, and make the code more succinct and readeable.
`<span>{{ isProvenanceGraphOpen ? 'Close' : 'Open' }}  Provenance Graph</span>`